app/controllers/api/v1/cars_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Api::V1::CarsController < ApplicationController
+  def index
+    @cars = Car.all
+
+    render json: { data: @cars }, status: :created
+  end
+
+  def create
+    @car = Car.new(cars_params)
+
+    if @car.save
+      render json: { message: 'Car created successfully', data: @car }, status: :created
+    else
+      render json: { message: 'There was an error' }, status: :unauthorized
+    end
+  end
+
+  def delete_car
+    car = Car.find(params[:car_id])
+
+    if car
+      car.reservations.delete_all
+      car.delete
```

Comment:
- [ ] Take a look [here](https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-dependent), you can use `dependent` in `has_many` to automatically destroy all reservations, wouldn't that be better?